### PR TITLE
Branch2

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
```
Comment (asamolion) [inline .idea/inspectionProfiles/Project_Default.xml]: Remove all of the .idea files.

File: books/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+
+
+class User(models.Model):
```
Comment (asamolion) [inline books/models.py]: Move the user model into a separate app. It should not be in books.

File: books/views.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    template_name = 'books/index.html'
+    context_object_name = 'latest_book_list'
+
+    def get_queryset(self):
+        """
+        Return the last five published questions (not including those set to be
+        published in the future).
+        """
+
+        return Book.objects.order_by('-id')[:5]
```
Comment (asamolion) [inline books/views.py]: Why is this statement needed?

File: books/views.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class DetailView(generic.DetailView):
```
Comment (asamolion) [inline books/views.py]: Why are you using class based views?

File: books/views.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        return Book.objects
+
+
+def login(request):
+    if request.POST['username'] == '' or request.POST['password'] == '':
+        return render(request, 'books/index.html', {
+            'error_message': "You didn't enter text in a field.",
+        })
+    if User.objects.filter(username__iexact=request.POST['username'], password=request.POST['password']).exists():
```
Comment (asamolion) [inline books/views.py]: This is will never login the user even with the correct password. Please rectify this.

Read about password hashing and correct this code.

File: books/views.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        return HttpResponse("You have successfully logged in!")
+    else:
+        return render(request, 'books/index.html', {
+            'error_message': "USERNAME/PASSWORD MISMATCH.",
+        })
+
+
+def signup(request):
```
Comment (asamolion) [inline books/views.py]: Either use class based views or functional views. Be consistent.

Comment (Ali-D-Akbar) [reply books/views.py]: we can actually use them at the same time as not every class based view can do all the needed work and exception handling.

File: books/views.py (hunk 5 above)
Comment (Ali-D-Akbar) [reply books/views.py]: custom user model has been added and it handles the hashing and stuff by itself.
so there will be no problem in login

**Pull request merged**